include/flatbuffers/flatbuffers.h

```diff
@@ -33,11 +33,30 @@
 #if __cplusplus <= 199711L && \
     (!defined(_MSC_VER) || _MSC_VER < 1600) && \
     (!defined(__GNUC__) || \
-      (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__ < 40603))
-  #error A C++11 compatible compiler is required for FlatBuffers.
+      (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__ < 40400))
+  #error A C++11 compatible compiler with support for the auto typing is required for FlatBuffers.
   #error __cplusplus _MSC_VER __GNUC__  __GNUC_MINOR__  __GNUC_PATCHLEVEL__
 #endif
 
+#if !defined(__clang__) && \
+    defined(__GNUC__) && \
+    (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__ < 40600)
+  // Backwards compatability for g++ 4.4, and 4.5 which don't have the nullptr and constexpr
+  // keywords. Note the __clang__ check is needed, because clang presents itself as an older GNUC
+  // compiler.
+  #ifndef nullptr_t
```

Comment:
Found this at https://en.wikibooks.org/wiki/More_C%2B%2B_Idioms/nullptr. It'll be easier to maintain the compatability easier with this idiom, since things like ```field_stack_.push_back(std::make_pair(val, nullptr));``` will no longer generate a compiler error because the type will be inferred. I also removed the member pointer cast since it's not used in this code anywhere. 

Please let me know if an attribution needs to be added somewhere.